doc/source/how-to/packaging.rst

```diff
@@ -363,6 +363,15 @@ optional dependencies:
      extras to replicate similar behavior.
    - Avoid specifying development-only dependencies in the main dependencies.
 
+   .. warning::
+
+      If you are using extras as a workaround for dependency groups, be cautious about the
+      potential for dependency conflicts. Extras are not designed to handle complex dependency
```

Comment:
I don't really understand the "Extras are not designed to handle complex dependency scenario." part. Maybe we could simplify the statement , e.g.

> If you are using extras as a workaround for dependency groups, be cautious that testing and documentation dependencies should not be user facing. Therefore, do not include inside the ``all`` target any of the testing or documentation dependencies (that is, ``pytest``, ``pytest-cov``, ``sphinx``, ``sphinx-autoapi``, and similar). You should create a separate extra (`dev`, `testing`, `documentation`, or similar) for such dependencies.

Reply:
By complex dependency scenarios I meant pinning down specific dev versions for packages... which is definitely complex for pip to resolve in combination with other libraries 😄 I'll rewrite it as you suggested, thanks @greschd!